meson.build

```diff
@@ -1654,6 +1654,17 @@ endforeach
 
 enable_sysusers = conf.get('ENABLE_SYSUSERS') == 1
 
+want = get_option('gshadow')
+if want.allowed()
+        have = conf.get('HAVE_GSHADOW_H') == 1
+else
+        have = false
+endif
+if want.enabled() and not have
+        error('support for shadow group is requested but could not find gshadow.h')
+endif
+conf.set10('ENABLE_GSHADOW', have)
```

Comment:
Should this not be `want`?  I'm not exactly sure how scoped variables work in Meson, but this seems like it would use the result of `get_option('xdg-autostart')` if `gshadow` is explicitly disabled as neither the `want.allowed()` nor `want.enabled()` would execute.